### ISSUE TITLE
[PATCH TermReadKey-2.14] Subroutine DynaLoader::dl_error    redefined on static build

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#1616]
Migrated from [rt.perl.org#2853](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=2853) (status was 'resolved')

Searchable as RT2853$

**Comment by p5pRT** [comment]
### From domo@computer.org

#### Created by domo@computer.org


\[Third shot\, in the hope that perlbug's up again\.\]

If perl is statically\-linked\, and in the absence of the patch below\,
CPAN squawks

$ perl \-MCPAN \-e shell
Subroutine DynaLoader&#8203;::dl\_error redefined at 
/usr/local/lib/perl5/5\.6\.0/powerpc\-machten/DynaLoader\.pm line 93\.

because ReadKey\.pm requires both AutoLoader and DynaLoader\.  The
patch does the same thing as
Change 5884 by gsar@&#8203;auger on 2000/03/22 18&#8203;:16&#8203;:45
did to IO\, and doesn't seem to stop anything working\.

<details><summary>Inline Patch</summary>

```diff
--- TermReadKey-2.14/ReadKey.pm~        Mon Mar 29 05:26:08 1999
+++ TermReadKey-2.14/ReadKey.pm Mon Mar 27 21:42:49 2000
@@ -208,10 +208,9 @@

  require Exporter;
  require AutoLoader;
-require DynaLoader;
  use Carp;

-@ISA = (Exporter, AutoLoader, DynaLoader);
+@ISA = (Exporter, AutoLoader);
  # Items to export into callers namespace by default
  # (move infrequently used names to @EXPORT_OK below)
  @EXPORT =  qw(
```
</details>


<details><summary>Perl Info</summary>

```

Flags:
     category=library
     severity=medium

Site configuration information for perl v5.6.0:

Configured by domo at Thu Mar 23 20:29:52 WET 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
   Platform:
     osname=machten, osvers=4.1.4, archname=powerpc-machten
     uname='machten ppp100 5 0.5 powerpc '
     config_args='-des'
     hint=recommended, useposix=true, d_sigaction=define
     usethreads=undef use5005threads=undef useithreads=undef 
usemultiplicity=undef
     useperlio=undef d_sfio=undef uselargefiles=define
     use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
   Compiler:
     cc='cc', optimize='-O2 -fomit-frame-pointer', gccversion=2.8.1
     cppflags='-DNOTDEF_MACHTEN -DREG_INFTY=2047 -I/usr/local/include'
     ccflags ='-DNOTDEF_MACHTEN -DREG_INFTY=2047 -I/usr/local/include'
     stdchar='char', d_stdstdio=undef, usevfork=true
     intsize=4, longsize=4, ptrsize=4, doublesize=8
     d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=8
     ivtype='long', ivsize=4, nvtype='double', nvsize=8, 
Off_t='off_t', lseeksize=4
     alignbytes=8, usemymalloc=y, prototype=define
   Linker and Libraries:
     ld='ld', ldflags =' -Xlstack=1048576 -L/usr/local/lib'
     libpth=/usr/local/lib /usr/lib
     libs=-lndbm -lgdbm -ldb -lm -lc
     libc=/usr/lib/libc.a, so=so, useshrplib=false, libperl=libperl.a
   Dynamic Linking:
     dlsrc=dl_none.xs, dlext=none, d_dlsymun=undef, ccdlflags=''
     cccdlflags='', lddlflags=''

Locally applied patches:



@INC for perl v5.6.0:
     /usr/local/lib/perl5/5.6.0/powerpc-machten
     /usr/local/lib/perl5/5.6.0
     /usr/local/lib/perl5/site_perl/5.6.0/powerpc-machten
     /usr/local/lib/perl5/site_perl/5.6.0
     /usr/local/lib/perl5/site_perl
     .


Environment for perl v5.6.0:
     HOME=/home/domo
     LANG (unset)
     LANGUAGE (unset)
     LD_LIBRARY_PATH=/usr/lib
     LOGDIR (unset)
 
PATH=/sbin:/usr/sbin:/home/domo/bin:/usr/local/bin:/usr/bin:/bin:/usr/X11/bin:/usr/libexec
     PERL_BADLANG (unset)
     SHELL=/bin/bash
-- 
Dominic Dunlop


```
</details>

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

This ticket can be closed&#8203;: patch 5947 and/or 7269 \(which appear to be 
the same patch on different branches\.  Or something\) fixed it\, along 
with two others\.
